### PR TITLE
Client Interface for Storage service

### PR DESCRIPTION
**Pull request by noobed-max**
- [x]  Develop save(key,value) function
- [x] Develop Retrieve(key) function 
- [x] Develop update(oldKey, newKey(optional), value) function
- [x] Develop delete(key) function
- [x] Unit Test Cases for each function 
#7 
#8 
#9 
#10 
#11

### CODE REVIEW

File: client/ciaos/client.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import requests
+
+
+def save(API_URL, key, value):
+    try:
+        data = {"key": key, "encoded_content": value}
```
Comment (Tejas-ChandraShekarRaju) [inline client/ciaos/client.py:6]: Why is this "enoded_content" and not value?

File: client/ciaos/client.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        response = requests.post(f"{API_URL}/upload/", data=data)
```
Comment (Tejas-ChandraShekarRaju) [inline client/ciaos/client.py:7]: Ideally this response should be returned. If in the future if the key is generated by us or any success/failure message the library user want's to know how will he know?

File: client/test/test.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#import the module here
```
Comment (Tejas-ChandraShekarRaju) [inline client/test/test.py:1]: Testing convention not followed.
https://fastapi.tiangolo.com/tutorial/testing/

**Review by Tejas-ChandraShekarRaju — CHANGES_REQUESTED** [review]
@noobed-max  Please make the following changes

**Comment by Tejas-ChandraShekarRaju** [comment]
@noobed-max  you have marked it will resolve update and delete but i don't see code for it.

File: client/ciaos/client.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            raise requests.HTTPError(response.text)
+
+    except requests.HTTPError as e:
+        print("Error:", e)
+
+
+def update(API_URL,key,value,new_key=None):
+    try:
+        data = {"new_key": new_key, "encoded_content": value}
+        response = requests.put(f"{API_URL}/update/?key={key}", data=data)
```
Comment (Tejas-ChandraShekarRaju) [inline client/ciaos/client.py:33]: @Ashish9738 where is the new key passed to the request?

File: client/ciaos/client.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        else:
+            raise requests.HTTPError(response.text)
+
+    except requests.HTTPError as e:
+        print("Error:", e)
+
+
+def delete(API_URL,key):
+    try:
+        response = requests.delete(f"{API_URL}/delete/?key={key}")
```
Comment (Tejas-ChandraShekarRaju) [inline client/ciaos/client.py:46]: return response.

File: client/ciaos/client.py (hunk 4 above)
Comment (Tejas-ChandraShekarRaju) [inline client/ciaos/client.py:33]: return response

**Review by Tejas-ChandraShekarRaju — CHANGES_REQUESTED** [review]

File: client/ciaos/client.py (hunk 4 above)
Comment (Ashish9738) [reply client/ciaos/client.py:33]: new key is passed with the data

File: client/test/test.py (hunk 3 above)
Comment (noobed-max) [reply client/test/test.py:1]: arent we testing the functions which are gonna hit fastapi 
But ig this conventions can be followed for server side

File: client/ciaos/client.py (hunk 2 above)
Comment (noobed-max) [reply client/ciaos/client.py:7]: isnt it handled by the code
 if response.status_code == 200:
            print("Update successful:", response.json())
        else:
            raise requests.HTTPError(response.text)

    except requests.HTTPError as e:
        print("Error:", e)

Comment (Tejas-ChandraShekarRaju) [reply client/ciaos/client.py:7]: You are the printing upon 200 where are you returning?

File: client/test/test.py (hunk 3 above)
Comment (Tejas-ChandraShekarRaju) [reply client/test/test.py:1]: Why? Conventions are there for a reason. It's not a guessing game please follow.

File: client/ciaos/client.py (hunk 2 above)
Comment (noobed-max) [reply client/ciaos/client.py:7]: ah yes i gotta modify except requests.HTTPError as e: print("Error:", e)

**Pull request closed without merging**